chalicelib_smaht/checks/audit_checks.py

```diff
@@ -90,21 +88,20 @@ def check_validation_errors(connection, **kwargs):
 def check_submitted_md5(connection, **kwargs):
     """ Check that any submitted md5s are consistent with the ones we generated """
     check = CheckResult(connection, 'check_submitted_md5')
-
-    search_url = 'search/?type=SubmittedFile&submitted_md5sum!=No+value&content_md5sum!=No+value&limit=500' \
-                 '&field=submitted_md5sum&field=content_md5sum'
-    results = ff_utils.search_metadata(search_url, key=connection.ff_keys)
-    atids = {result['@id'] for result in results if result['submitted_md5sum'] != result['content_md5sum']}
+    search_stem = 'search/?type=SubmittedFile&submitted_md5sum!=No+value&md5sum!=No+value'
```

Comment:
Not sure I understand the check. Does this check the same files over and over again? If so, I think you need the limit that your removed, so that the old ones eventually fall out of scope.

Reply:
Good point - that is the way that it was but it was actually comparing the wrong md5 fields and had another bug.  In theory either value could be updated manually even if unlikely.  But I agree silly to check all the files every time.  But limiting to just a certain amount of search results is also not that useful - so I retrieve the latest primary check result and look for files modified from that date - unless an explicit date is provided as a kwarg.  This doesn't track (in the current result) if there were cases from a previous result that were not 'fixed' so maybe still not ideal but so far there don't seem to be any cases so for now I think that is OK.